### DISCUSSION TITLE
Auth Issues with Sveltekit

### DISCUSSION THREAD

**Discussion by akiarostami** [directus/directus#17040] [Archive]
Hi,

I'm having a lot of issues with Sveltekit integration / authentication. auth.login doesn't set cookies (even though I speicifcally set  `mode: 'cookie'`. 

Your Sveltekit example is very old. Now that Sveltekit 1.0 is out, is there a chance to have a sample app with proper authentication code?

**Comment by Lotta-Llamas** [comment]
Not sure if you are still seeking a solution for this, but here is what I came up with:
```			
const directusClient = await client.with(authentication());
await directusClient.login(email, password, {});
const token = (await directusClient.getToken()) as string;

cookies.set('client_token', token, {
  path: '/',
  httpOnly: true,
  sameSite: 'strict',
  maxAge: 60 * 60 * 24 * 7 // one week
 });
```
This happens on the +page.server.ts action hook

Then when making subsequent requests that need the authentication, you can set that token in the client via `directusClient.setToken(token)`

I hope this helps 🤞
